src/sql/db_connection_pool/dbconnection/mysqlconn.rs

```diff
@@ -72,13 +99,10 @@ impl<'a> AsyncDbConnection<Conn, &'a (dyn ToValue + Sync)> for MySQLConnection {
         let mut conn = self.conn.lock().await;
         let conn = &mut *conn;
 
-        // we don't use SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '{}' AND TABLE_SCHEMA = '{}'
-        // as table_reference don't always have schema specified so we need to extract schema/db name from connection properties
-        // to ensure we are querying information for correct table
-        let columns_meta_query =
-            format!("SHOW COLUMNS FROM {}", table_reference.to_quoted_string());
-
-        let columns_meta: Vec<Row> = match conn.exec(&columns_meta_query, Params::Empty).await {
+        let columns_meta: Vec<Row> = match conn
+            .exec(Self::column_meta_query_for(table_reference), Params::Empty)
```

Comment:
Instead of building query as string, it would be safer to use prepared statement and pass the table name as value, something like `tx.exec("INSERT INTO tmp (a) VALUES (?)", ("foo",))?;`

Reply:
The current approach could be vulnerable to SQLi, user can pass in name like this: 
```
fake_name`; DELETE TABLE secrets;--
```
and it will be passed in as query param to perform bad operations
```
SHOW COLUMNS FROM `catalog`.`schema`.`fake_name`; DELETE TABLE secrets;--`
```

Reply:
From what I found, `SHOW` doesn't support prepared statements in that manner.

Reply:
@Sevenannn  - good catch. This is DF reference implementation - should we just follow the same approach.



```
/// Wraps identifier string in double quotes, escaping any double quotes in
/// the identifier by replacing it with two double quotes
///
/// e.g. identifier `tab.le"name` becomes `"tab.le""name"`
pub fn quote_identifier(s: &str) -> Cow<str> {
    if needs_quotes(s) {
        Cow::Owned(format!("\"{}\"", s.replace('"', "\"\"")))
    } else {
        Cow::Borrowed(s)
    }
}

/// returns true if this identifier needs quotes
fn needs_quotes(s: &str) -> bool {
    let mut chars = s.chars();

    // first char can not be a number unless escaped
    if let Some(first_char) = chars.next() {
        if !(first_char.is_ascii_lowercase() || first_char == '_') {
            return true;
        }
    }

    !chars.all(|c| c.is_ascii_lowercase() || c.is_ascii_digit() || c == '_')
}
```

```
pub fn to_quoted_string(&self) -> String {
        match self {
            TableReference::Bare { table } => quote_identifier(table).to_string(),
            TableReference::Partial { schema, table } => {
                format!("{}.{}", quote_identifier(schema), quote_identifier(table))
            }
            TableReference::Full {
                catalog,
                schema,
                table,
            } => format!(
                "{}.{}.{}",
                quote_identifier(catalog),
                quote_identifier(schema),
                quote_identifier(table)
            ),
        }
    }
```

Reply:
I don't think we need all of this?

Reply:
The `parse_multipart_identifier` used on the Spice side seem to be pretty strict on parsing the catalog / schema / table name as it doesn't allow extraneous tokens so seems that this wouldn't cause SQLi in Spice. I wonder if the method should be updated here for people merely uses datafusion-table-provider repo.

Reply:
Yes, but it doesn't work for `SHOW COLUMNS` in MySQL.